tools/osbuild-depsolve-dnf5

```diff
@@ -0,0 +1,500 @@
+                "repo_id": package.get_repo_id(),
+                "path": package.get_location(),
+                "remote_location": remote_location(package),
+                "checksum": (
+                    f"{package.get_checksum().get_type_str()}:"
+                    f"{package.get_checksum().get_checksum()}"
+                )
+            })
+
+        return sorted(dependencies, key=lambda x: x["path"])
+
+
+def setup_cachedir(request):
+    arch = request["arch"]
+    # If dnf-json is run as a service, we don't want users to be able to set the cache
+    cache_dir = os.environ.get("OVERWRITE_CACHE_DIR", "")
+    if cache_dir:
+        cache_dir = os.path.join(cache_dir, arch)
+    else:
+        cache_dir = request.get("cachedir", "")
+
+    if not cache_dir:
+        return "", {"kind": "Error", "reason": "No cache dir set"}
+
+    return cache_dir, None
+
+
+def solve(request, cache_dir):
+    command = request["command"]
+    arch = request["arch"]
+    module_platform_id = request["module_platform_id"]
+    arguments = request["arguments"]
+
+    # Gather up all the exclude packages
+    exclude_pkgs = []
+    transactions = arguments.get("transactions")
+    if transactions:
+        for t in transactions:
+            if t.get("exclude-specs") is None:
+                continue
+            exclude_pkgs += t.get("exclude-specs")
+
+    with tempfile.TemporaryDirectory() as persistdir:
+        try:
+            solver = Solver(
+                arguments["repos"],
+                module_platform_id,
+                persistdir,
+                cache_dir,
+                arch,
+                exclude_pkgs
+            )
+            if command == "dump":
+                result = solver.dump()
+            elif command == "depsolve":
+                result = solver.depsolve(transactions)
+            elif command == "search":
+                result = solver.search(arguments.get("search", {}))
+        except TransactionError as e:
+            printe("error depsolve")
+            return None, {
+                "kind": "DepsolveError",
+                "reason": f"There was a problem with depsolving: {e}"
+            }
+        except RepoError as e:
+            printe("error repository setup")
+            return None, {
+                "kind": "RepoError",
+                "reason": f"There was a problem reading a repository: {e}"
+            }
+        except Exception as e:
+            import traceback
+            printe("error traceback")
+            return None, {
+                "kind": type(e).__name__,
+                "reason": str(e),
+                "traceback": traceback.format_exc()
+            }
+    return result, None
+
+
+def printe(*msg):
+    print(*msg, file=sys.stderr)
+
+
+def fail(err):
+    printe(f"{err['kind']}: {err['reason']}")
+    print(json.dumps(err))
+    sys.exit(1)
+
+
+def respond(result):
+    print(json.dumps(result))
+
+
+def validate_request(request):
```

Comment:
Same question as above if code could be shared between this and osbuild-depsolve-dnf as this and main look identical (from a quick glance)

Reply:
Same answer ;)